spec/lib/params_sanitizer_spec.rb

```diff
@@ -0,0 +1,160 @@
+    context "when xml is nil" do
+      let(:params) { { "xml" => nil } }
+      subject { ParamsSanitizer.new(params) }
+
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is empty" do
+      let(:params) { { "xml" => "" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is not string" do
+      let(:params) { { "xml" => 123 } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+
+    context "when xml is string" do
+      let(:params) { { "xml" => "<xml></xml>" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return hash" do
+        expect(subject.add_xml).to eq(nil)
+      end
+    end
+  end
+
+  describe "add_metadata_version" do
+    context "when params is nil" do
+      let(:params) { { "version_info" => nil } }
+      subject { ParamsSanitizer.new(params) }
+
+      it "should return nil" do
+        subject.add_metadata_version(params)
+        expect(subject.get_params[:version_info]).to eq(nil)
+      end
+    end
+
+    context "when params is empty" do
+      let(:params) { { "version_info" => "" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        subject.add_metadata_version(params)
+        expect(subject.get_params[:version_info]).to eq(nil)
+      end
+    end
+
+    context "when params is not hash" do
+      let(:params) { { "version_info" => 123 } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        subject.add_metadata_version(params)
+        expect(subject.get_params[:version_info]).equal?(123)
+      end
+    end
+  end
+
+
+  describe "add_random_id" do
+    context "when params is nil" do
+      let(:params) { { "prefix" => nil, "doi" => nil } }
+      subject { ParamsSanitizer.new(params) }
+
+      it "should return nil" do
+        subject.add_random_id
+        expect(subject.get_params[:doi]).to eq(nil)
+      end
+    end
+
+    context "when params is empty" do
+      let(:params) { { "prefix" => "", "doi" => "" } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        subject.add_random_id
+        expect(subject.get_params[:doi]).to eq(nil)
+      end
+    end
+
+    context "when params is not hash" do
+      let(:params) { { "prefix" => 123, "doi" => 123 } }
+      subject { ParamsSanitizer.new(params) }
+      it "should return nil" do
+        subject.add_random_id
+        expect(subject.get_params[:doi]).to eq(nil)
+      end
+    end
+
+    context "when params is hash" do
+      subject { ParamsSanitizer.new({ prefix: "10.1233", doi: "" }) }
+
+      it "adds correctly random doi" do
+        subject.add_random_id()
+        expect(subject.get_params[:doi]).to start_with("10.1233")
```

Comment:
I think because the test environment generates predictable DOIs (per comment and code here: https://github.com/datacite/lupo/pull/886/files#diff-1f6e1cee6b8f670353834f39818e744200a5edc30f509ed1655ac918b7295a0cR430) we could increase this assertion to the whole DOI rather than just the prefix

Reply:
I think that would be extra